forge-game/src/main/java/forge/game/ability/effects/TapEffect.java

```diff
@@ -48,7 +48,7 @@ public void resolve(SpellAbility sa) {
                 if (tgtC.isUntapped() && remTapped || alwaysRem) {
                     card.addRemembered(tgtC);
                 }
-                tgtC.tap(true);
+                tgtC.tap(true, sa);
```

Comment:
what about funky cards like _Raiding Party_?
While you control the SA you're not always the tapper... :/

Reply:
for such cases we probably need to wait for CR to fix that later